source/base/data_out_base.cc

```diff
@@ -1427,6 +1427,78 @@ namespace
 
 
 
+  // Separate these out to avoid an internal compiler error with intel 17
+  namespace internal
+  {
+    template <int dim>
+    std::array<unsigned int, GeometryInfo<dim>::vertices_per_cell>
+    order_cell(unsigned int /*start*/,
```

Comment:
Could you add a comment to one of these functions? The argument list doesn't give away what the function actually does, but it doesn't seem to have much to do with ordering cells ;-)